asciidoc/courses/llm-fundamentals/modules/2-vectors-semantic-search/lessons/1-semantic-search/lesson.adoc

```diff
@@ -0,0 +1,68 @@
+= Vectors & Semantic Search
+
+// TODO: Rewrite
+In the last lesson, we discussed the importance of grounding to improve LLM accuracy and the concept of **Retrieval Augmented Generation** (RAG).
+RAG involves providing additional information along with a prompt to help the LLM form a response.
+
+One of the big challenges of RAG is understanding what the user is asking for, and to surface the right information to pass to the LLM.
+
+
+== Semantic Search vs. Traditional Search
+
+At its core, semantic search aims to understand the intent and contextual meaning of search phrases, rather than just focusing on individual keywords.
+Traditional search often depends on exact match keywords or proximity-based algorithms to return results.
+
+For example, if you input "apple" in a traditional search, you might predominantly get results about the fruit.
+However, in a semantic search, the engine tries to gauge the context: Are you searching about the fruit, the tech company, or something else?
+
+The results are hence tailored not just based on the term, but the perceived intent behind it.
+
+== Vectors and embeddings
+
+
+In the realm of natural language processing (NLP) and machine learning, words or phrases are often converted into numerical representations known as **vectors**.
+Each dimension in a vector can represent a particular semantic aspect of the word or phrase.
+With advanced models, these embeddings also contain contextual information.
+This means the vector for a word can change based on its surrounding context. For instance, the word _bank_ will have a different vector in _river bank_ than in _savings bank_.
+Semantic search systems can use these contextual embeddings to better understand user intent.
+
```

Comment:
perhaps provide an example of vector operations, either the old one of king-man+woman = queen or something less used.